### PR TITLE
parameterize runtime memory via machine_mem_gb input

### PR DESCRIPTION
**Pull request by tomkinsc**
parameterize runtime memory via the optional input `machine_mem_gb`, defaulting to the same hard-coded values before if `machine_mem_gb` is null. This also raises the memory value in a few places to be closer to the resources of the specified DNAnexus instances.

### CODE REVIEW

File: pipes/WDL/tasks/tasks_demux.wdl [hunk 1]
```diff
@@ -246,7 +249,7 @@ task illumina_demux {
 
   runtime {
     docker: "${docker}"
-    memory: "200 GB"
+    memory: select_first([machine_mem_gb, 240]) + " GB"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_demux.wdl:252]: On GCE, an `n1-highmem-32` is 208GB of RAM, which motivated this original value (and I think we never saw the high water mark above 160-180GB) -- setting it to 240 will probably cause it to go for an `n1-standard-64` or something on those platforms. Can we keep it to 200 to fit a better common denominator between EC2 and GCE?

File: pipes/WDL/tasks/tasks_metagenomics.wdl [hunk 2]
```diff
@@ -84,7 +85,7 @@ task krakenuniq {
 
   runtime {
     docker: "${docker}"
-    memory: "240 GB"
+    memory: select_first([machine_mem_gb, 240]) + " GB"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_metagenomics.wdl:88]: this is a separate issue, but Simon's new krakenuniq database is now 282GB! Maybe we should bump this to 320GB or so (and explicitly ask dxWDL for a `mem3_ssd1_v2_x48`).

File: pipes/WDL/tasks/tasks_demux.wdl (hunk 1 above)
Comment (tomkinsc) [reply pipes/WDL/tasks/tasks_demux.wdl]: Sure thing; I changed it back. The increase was in response to demux failure for a NovaSeq S4 on Terra, but with parameterization it'll be straightforward to re-launch with more memory.

File: pipes/WDL/tasks/tasks_metagenomics.wdl (hunk 2 above)
Comment (tomkinsc) [reply pipes/WDL/tasks/tasks_metagenomics.wdl]: Done.

**Pull request merged**